src/main/java/com/seals/camplanner/commons/services/BaseService.java

```diff
@@ -0,0 +1,42 @@
+package com.seals.camplanner.commons.services;
+
+import com.seals.camplanner.commons.models.BaseEntity;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public abstract class BaseService<T extends BaseEntity> implements IBaseService<T> {
+
+    protected abstract JpaRepository<T, Long> getRepository();
+
+    public abstract String getEntityName();
```

Comment:
@Icaruspooky Another way of doing this would be to have them as class fields and receive them as parameters.

Considerations about using abstract methods:
- They cant be used in the constructor
- I feel they are harder to read
- Forces us to have a getter for the repository, although its protected this might not be ideal.

Considerations about having them as class fields:
- Field would be duplicated on the parent and child class, not a memory problem as they would use the same reference, but weird.
- @RequiredArgsConstructor would no longer satisfy our needs, we would have to write the constructor
- We could also have the repository only on the parent have the child access it through a getter, it would be basically the same thing as the former option, but it would be usable on the constructor.

Do you have an opinion on this?